OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/WorkshopControllerTests.cs

```diff
@@ -533,14 +533,35 @@ public async Task UpdateWorkshop_WhenDtoAvailableSeatsIsInvalid_ShouldReturnBadR
         workshopUpdateDto.ProviderId = provider.Id;
         providerServiceMoq.Setup(x => x.IsBlocked(It.IsAny<Guid>())).ReturnsAsync(false);
         providerServiceMoq.Setup(x => x.GetByUserId(It.IsAny<string>(), It.IsAny<bool>())).ReturnsAsync(provider);
+        workshopServiceMoq.Setup(x => x.GetById(It.IsAny<Guid>(), It.IsAny<bool>())).ReturnsAsync(workshop);
         workshopServiceMoq.Setup(x => x.IsAvailableSeatsValid(It.IsAny<uint?>(), It.IsAny<WorkshopDto>())).Returns(false);
 
         // Act
         var result = await controller.Update(workshopUpdateDto).ConfigureAwait(false) as ObjectResult;
 
         // Assert
         providerServiceMoq.VerifyAll();
-        workshopServiceMoq.Verify(x => x.Update(It.IsAny<WorkshopBaseDto>()), Times.Never);
+        workshopServiceMoq.VerifyAll();
+        Assert.IsNotNull(result);
+        Assert.AreEqual(BadRequest, result.StatusCode);
+    }
+
+    [Test]
+    public async Task UpdateWorkshop_WhenWorkshopEntityNotExist_ShouldReturnBadRequestObjectResult()
+    {
+        // Arrange
+        workshopUpdateDto.ProviderId = provider.Id;
+        var workshopDto = null as WorkshopDto;
+        providerServiceMoq.Setup(x => x.IsBlocked(It.IsAny<Guid>())).ReturnsAsync(false);
+        providerServiceMoq.Setup(x => x.GetByUserId(It.IsAny<string>(), It.IsAny<bool>())).ReturnsAsync(provider);
+        workshopServiceMoq.Setup(x => x.GetById(It.IsAny<Guid>(), It.IsAny<bool>())).ReturnsAsync(workshopDto);
```

Comment:
(Test design) Please avoid using `It.IsAny` when you know what data are expected. This way your test code is more clear (as instead of abstract `It.IsAny<Guid>()` you'll have specific `provider.Id`) and more precise in testing.

Reply:
Fixed. Thank you!